### PR TITLE
Version increment 2.4

### PR DESCRIPTION
**Pull request by phestecAMZN**
Signed-of-by: Chris Hesterman <phestech@amazon.com>

### Description
Increments version to 2.4

Signed-off-by: Chris Hesterman <phestech@amazon.com>

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: cypress/integration/rollups_spec.js [hunk 1]
```diff
@@ -21,6 +21,9 @@ describe("Rollups", () => {
     // // Click on "Sample data" tab
     // cy.contains("Add data").click({ force: true });
```
Comment (deshsidd) [inline cypress/integration/rollups_spec.js:22]: Do we need the duplicate code here?

Comment (phestecAMZN) [reply cypress/integration/rollups_spec.js:22]: This is a bit of a flaky test, sometimes needs the commented code to pass, sometimes doesn't.

Comment (phestecAMZN) [reply cypress/integration/rollups_spec.js:22]: oh I see, yes I'll get rid of the commented, since it is being used.

Comment (phestecAMZN) [reply cypress/integration/rollups_spec.js]: Done.

**Review by amsiglan — DISMISSED** [review]

File: .github/workflows/cypress-workflow.yml [hunk 2]
```diff
@@ -7,8 +7,8 @@ on:
     branches:
       - "*"
 env:
-  OPENSEARCH_DASHBOARDS_VERSION: '2.3.0'
-  OPENSEARCH_VERSION: '2.3.0-SNAPSHOT'
+  OPENSEARCH_DASHBOARDS_VERSION: '2.4.0'
```
Comment (bowenlan-amzn) [inline .github/workflows/cypress-workflow.yml:10]: change to 2.x

**Review by AWSHurneyt — APPROVED** [review]

**Pull request merged**